### DISCUSSION TITLE
Setting up vercel wildcard domains with namecheap: "Invalid configuration"

### DISCUSSION THREAD

**Discussion by AIEngKenn** [vercel/community#1504] [Help]
### Summary

Sorry (if this question seems to be silly, currently learning how to implement the feature),

1) I added a domain *.example.com
2) Added the intended nameservers on Namecheap
3) "Invalid Configuration" on Vercel
4) Got stuck and contacted the Namecheap team, they said everything was Ok on their side :(
5) Here I am, Need some help :(

Thanks

**Comment by swarnava** [comment]
Hi, What is the domain name?

**Reply by AIEngKenn** [reply]
EarlyAccess.to

**Comment by swarnava** [comment]
Looks like your NameServer hasn't propagated to Vercel yet: https://dnschecker.org/#NS/EarlyAccess.to

**Reply by AIEngKenn** [reply]
Well, I am not sure I understand.

but the domain works fine. (EarlyAccess.to & app.earlyaccess.to). 

All is working perfectly. The problem is: it doesn’t work when I try to add the *.earlyaccess.to domain… 

I’ve already added the intended name servers to Namecheap. But it still doesn’t work. 

Any help?

**Reply by swarnava** [reply]
Hi Kenn,

EarlyAccess.to and app.earlyaccess.to are working because the DNS record are set with desired value set in your Domain (Namecheap dashboard:

- https://dnschecker.org/#A/EarlyAccess.to
- https://dnschecker.org/#CNAME/app.EarlyAccess.to

Howver the NameServer hasn't switched to Vercel which is essentially require to make wildcard domain work. You can use https://dnschecker.org/#NS/EarlyAccess.to or any online DNS tool to check this. You need to reach out to Namecheap server and ask why the correct Nameserver is not propagating.

**Reply by AIEngKenn** [reply]
just talked with them, and here is the response (after more than 20 minutes of several tries): 

"you can try contacting Vercel and let them know, that Namecheap does not allow using NS records for wildcard subdomains. The NS record can be added only for the exact subdomain name (for example domain.EarlyAccess.to or example.EarlyAccess.to)."

Any help?

**Reply by smaeda-ks** [reply]
@KennStack01 you don't need to create NS records on the wildcard alias on Namecheap. What you needed to follow instead is to change your name servers to Vercel:
https://www.namecheap.com/support/knowledgebase/article.aspx/767/10/how-to-change-dns-for-a-domain/

^^ By selecting the `Custom DNS` option, you should be able to change the default name servers.

Our name servers are:
```
ns1.vercel-dns.com
ns2.vercel-dns.com
```

- https://vercel.com/docs/concepts/projects/domains/troubleshooting#common-nameserver-issues

Once it's done, you will be able to manage your DNS records in the Vercel UI. But please be cautious with this operation. Namecheap also mentions:
> NOTE: If you change nameservers from Custom nameservers to our default ones or vice versa, the host records will not be added to the new nameservers automatically. Please make sure to save all the needed settings and set them up manually after the nameservers switch.

So basically, before changing the name servers, you need to manually "sync" DNS records between Namecheap and Vercel. You can export all zones from Namecheap and import it via:
https://vercel.com/dashboard/domains/earlyaccess.to
Please make sure to complete this step beforehand.

**Reply by shansrik** [reply]
Did you ever get this working @KennStack01? I just tried this and no luck.

**Reply by AIEngKenn** [reply]
No luck! Nothings works. Vercel is so big now, they're very slow and unhelpful these days... it creates frustrations. Better to find another faster solution, especially with their App router thing, it's going to be harder to get help. 

...unless they've updated the solution, better to find another tool or platform. @shansrik

**Reply by shansrik** [reply]
Ah that's frustrating. @smaeda-ks @swarnava are you able to help? On my end, I can't get past the step of clicking 'Enable DNS'. Is there a way to test this again without app going down while nameserver propagates?
